frame/executive/src/lib.rs

```diff
@@ -216,10 +216,17 @@ where
 		// any initial checks
 		Self::initial_checks(&block);
 
+		sp_io::extensions::start_verification_extension();
+
 		// execute extrinsics
 		let (header, extrinsics) = block.deconstruct();
 		Self::execute_extrinsics_with_book_keeping(extrinsics, *header.number());
 
+		if !sp_io::crypto::batch_verify() {
```

Comment:
can you elaborate a bit more on how this affects the runtime? what does this call do? (this code line is kinda arbitrary, I am just looking for an excuse to extract more info here :D)

Also 

- Is this for validation or execution, or both? 
- I assume the details are out of scope, but generally speaking, how is this batching improving the speed? Is it parallel or is it due to some math property?

Reply:
Yes, this probably should be wrapped in some structure with drop handling / consuming finaliser.

It is for execution only, for validation it does not make much sense.

During the block execution, all signature verifications requests are added to the batch on native side (see `batch_push_ed25519` / `batch_push_sr25519`) instead of being verified immediately.

Then this  `sp_io::crypto::batch_verify()` is called, and all accumulated signatures are batch-verified (and/or parallelized - coming soon). With current approach, it is also possible to actually verify signatures in parallel on native side while runtime still checks the rest of the block transaction correctness.

Reply:
> I assume the details are out of scope, but generally speaking, how is this batching improving the speed? Is it parallel or is it due to some math property?

I'll add detailed description soon-ish for this and other unclear moments also